From 2169ec8a8744149b8d7a9518568912109949003e Mon Sep 17 00:00:00 2001
From: Alexander Golubev <fatzer2@gmail.com>
Date: Fri, 1 Aug 2025 18:32:41 +0300
Subject: [PATCH 24/26] [ffmpeg] set default for rc_strategy to 0

Value 2 were never correct or supported by ffmpeg. And ffmpeg-3 started
to enforce this option to be in range [0;1].

Note: technically this option should be passed only to mpeg2 codec, but
that's  too much bother to find how to do it.

Signed-off-by: Alexander Golubev <fatzer2@gmail.com>
---
 encode/encode_lavc.c | 4 ++--
 export/ffmpeg_cfg.c  | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/encode/encode_lavc.c b/encode/encode_lavc.c
index 2211934d..64bb1735 100644
--- a/encode/encode_lavc.c
+++ b/encode/encode_lavc.c
@@ -995,7 +995,7 @@ static void tc_lavc_config_defaults(TCLavcPrivateData *pd)
     pd->ff_vcontext.b_sensitivity           = 40;
     pd->ff_vcontext.brd_scale               = 0;
     pd->ff_vcontext.bidir_refine            = 0;
-    pd->confdata.rc_strategy                = 2;
+    pd->confdata.rc_strategy                = 0;
     pd->ff_vcontext.b_quant_factor          = 1.25;
     pd->ff_vcontext.i_quant_factor          = 0.8;
     pd->ff_vcontext.b_quant_offset          = 1.25;
@@ -1196,7 +1196,7 @@ static int tc_lavc_read_config(TCLavcPrivateData *pd,
         { "vrc_maxrate", PAUX(rc_max_rate), TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 0, 24000000 },
         { "vrc_minrate", PAUX(rc_min_rate), TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 0, 24000000 },
         { "vrc_buf_size", PAUX(rc_buffer_size), TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 4, 24000000 },
-        { "vrc_strategy", PAUX(rc_strategy), TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 0, 2 },
+        { "vrc_strategy", PAUX(rc_strategy), TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 0, 1 },
         { "vb_qfactor", PCTX(b_quant_factor), TCCONF_TYPE_FLOAT, TCCONF_FLAG_RANGE, -31.0, 31.0 },
         { "vi_qfactor", PCTX(i_quant_factor), TCCONF_TYPE_FLOAT, TCCONF_FLAG_RANGE, -31.0, 31.0 },
         { "vb_qoffset", PCTX(b_quant_offset), TCCONF_TYPE_FLOAT, TCCONF_FLAG_RANGE, 0.0, 31.0 },
diff --git a/export/ffmpeg_cfg.c b/export/ffmpeg_cfg.c
index 8ce82236..bf9c3d3d 100644
--- a/export/ffmpeg_cfg.c
+++ b/export/ffmpeg_cfg.c
@@ -46,7 +46,7 @@ float lavc_param_vi_qoffset = 0.0;
 int lavc_param_vmax_b_frames = 0;
 //int lavc_param_keyint = -1;
 //int lavc_param_vpass = 0;
-int lavc_param_vrc_strategy = 2;
+int lavc_param_vrc_strategy = 0;
 int lavc_param_vb_strategy = 0;
 int lavc_param_luma_elim_threshold = 0;
 int lavc_param_chroma_elim_threshold = 0;
@@ -149,7 +149,7 @@ TCConfigEntry lavcopts_conf[]={
     {"vb_qfactor", &lavc_param_vb_qfactor, TCCONF_TYPE_FLOAT, TCCONF_FLAG_RANGE, -31.0, 31.0},
     {"vmax_b_frames", &lavc_param_vmax_b_frames, TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 0, INT_MAX},
 //    {"vpass", &lavc_param_vpass, TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 0, 2},
-    {"vrc_strategy", &lavc_param_vrc_strategy, TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 0, 2},
+    {"vrc_strategy", &lavc_param_vrc_strategy, TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 0, 1},
     {"vb_strategy", &lavc_param_vb_strategy, TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 0, 10},
     {"vb_qoffset", &lavc_param_vb_qoffset, TCCONF_TYPE_FLOAT, TCCONF_FLAG_RANGE, 0.0, 31.0},
     {"vlelim", &lavc_param_luma_elim_threshold, TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, -99, 99},
-- 
2.49.1

